File: LogicalThinkingPatterns/IncreasingNumberTrianglePattern/IncreasingNumberTriangle.ts
class IncreasingNumberTriangle {
  printPattern(n: number): void {
    let count = 1;
    for (let i = 0; i < n; i++) {
      let pattern: string = "";
      for (let j = 0; j <= i; j++) {
        pattern += count + " ";
        count++;
      }
      console.log(pattern);
    }
  }
}

const triangle = new IncreasingNumberTriangle();
triangle.printPattern(5);

File: LogicalThinkingPatterns/IncreasingNumberTrianglePattern/IncreasingNumberTriangle.py
class IncreasingNumberTriangle:
    def printPattern(self, n):
        count = 1;
        for i in range(0, n):
            pattern = ""
            for j in range(0, i+1):
                pattern += str(count) + " "
                count += 1
            print(pattern)
        
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

triangle = IncreasingNumberTriangle()
triangle.run()